### PR TITLE
HL-246

### PR DESCRIPTION
**Pull request by OptimusOpus**
[Link to ticket](https://dltx.atlassian.net/jira/software/projects/HL/boards/82?selectedIssue=HL-246)

### CODE REVIEW

**Review by brettvniekerk — APPROVED** [review]

**Review by bitcoinbrisbane — APPROVED** [review]

File: src/hooks/market/useMarkets.ts [hunk 1]
```diff
@@ -2,18 +2,20 @@ import { useContractRead, useContractReads } from "wagmi";
...
 import api from "../../apis/Api";
 
-const registryContractAddress = process.env.VITE_REGISTRY_CONTRACT;
-if (!registryContractAddress)
-  throw new Error("No VITE_REGISTRY_CONTRACT provided");
+const useMarketAddressesFromContract = () => {
+  const config = useConfig();
+  const registryContractAddress = config?.addresses?.registry;
+  if (!registryContractAddress)
+    throw new Error("No REGISTRY_CONTRACT provided");
```
Comment (brettvniekerk) [inline src/hooks/market/useMarkets.ts:12]: we dont want to throw if there is no address provided -- `undefined` happens if the config is loading, i think you can just pass an undefined address into the `registryContract` object?

File: src/hooks/vault/useVaults.ts [hunk 2]
```diff
@@ -3,17 +3,19 @@ import registryContractJson from "../../abi/Registry.json";
...
+import { useConfig } from "../../providers/Config";
 
-const registryContractAddress = process.env.VITE_REGISTRY_CONTRACT;
-if (!registryContractAddress)
-  throw new Error("No VITE_REGISTRY_CONTRACT provided");
+const useVaultAddresesFromContract = () => {
+  const config = useConfig();
+  const registryContractAddress = config?.addresses?.registry;
+  if (!registryContractAddress)
+    throw new Error("No REGISTRY_CONTRACT provided");
```
Comment (brettvniekerk) [inline src/hooks/vault/useVaults.ts:12]: same here

File: src/hooks/market/useMarkets.ts (hunk 1 above)
Comment (brettvniekerk) [reply src/hooks/market/useMarkets.ts:12]: alternatively, you could pass `config` in as a prop and load it in the parent component

**Review by brettvniekerk — DISMISSED** [review]

File: src/hooks/vault/useVaults.ts [hunk 3]
```diff
@@ -51,14 +14,8 @@ const useVaultAddresesFromAPI = () => {
...
 
-const shouldUseAPI = process.env.VITE_REST_FOR_VAULTS;
-
 const useVaults = () => {
-  if (shouldUseAPI) {
-    const { vaultAddresses } = useVaultAddresesFromAPI();
-    return { vaultAddresses };
-  }
-  const { vaultAddresses } = useVaultAddresesFromContract();
+  const { vaultAddresses } = useVaultAddresesFromAPI();
```
Comment (bazmatic) [inline src/hooks/vault/useVaults.ts:18]: Whoopsies!
```suggestion
  const { vaultAddresses } = useVaultAddressesFromAPI();
```

**Review by bazmatic — CHANGES_REQUESTED** [review]
Spelling

**Review by pondpiu — APPROVED** [review]

**Review by bitcoinbrisbane — APPROVED** [review]

**Review by bazmatic — APPROVED** [review]

**Pull request merged**